src/components/AttendeeForm.tsx

```diff
@@ -0,0 +1,259 @@
+'use client'
+
+import { comment } from 'postcss'
+import type { ChangeEvent } from 'react'
+import { useState } from 'react'
+
+import { usePocketBase } from '../contexts/PocketBaseContext'
+import { FileInput, Input, SelectInput } from './FormInputs'
+
+type Props = {
+  tournamentId: string
+  onSuscribeOk: () => void
+}
+const AttendeeForm: React.FC<Props> = ({ tournamentId, onSuscribeOk }) => {
+  const validate = (newInputs: Inputs): Errors => {
+    const newErrors: Errors = {}
+
+    //riotId
+    const riotIdRegex = /^.{4,16}#[a-zA-Z0-9]{3,5}$/
+    if (!riotIdRegex.test(newInputs.riotId)) {
+      newErrors.riotId = 'Please enter a valid Riot ID'
+    }
+    //currentElo
+    // const rankRegex = /^(iron|bronze|silver|gold|platinium|emerald|diamond)\s*[1-4]$/
+    // if (!rankRegex.test(newInputs.currentElo)) {
+    //   newErrors.currentElo =
+    //     'Please enter a valid Elo (iron|bronze|silver|gold|platinium|emerald|diamond)[1-4]'
+    // }
+    //comment
+    if (comment.length > 50) {
+      newErrors.comment = 'Please a comment under 50 characters'
+    }
+    //Role
+    const roleRegex = /^(adc|jungle|mid|support|top)$/
+    if (!roleRegex.test(newInputs.role)) {
+      newErrors.role = 'Please enter a valid role (adc|jungle|mid|support|top)'
+    }
+
+    //Birthplace
+    if (newInputs.birthPlace === '') {
+      newErrors.birthPlace = 'Please enter a birthplace'
+    }
+
+    //Validate avatar
+
+    if (newInputs.avatar === null) {
+      newErrors.avatar = 'Please choose an avatar'
+    }
+
+    return newErrors
+  }
+
+  type Inputs = {
+    riotId: string
+    currentElo: string
+    comment: string
+    role: string
+    championPool: string
+    birthPlace: string
+    avatar: File | null
+  }
+  const [inputs, setInputs] = useState<Inputs>({
+    riotId: '',
+    currentElo: 'Fer',
+    comment: '',
+    role: 'adc',
+    championPool: 'One trick poney',
+    birthPlace: '',
+    avatar: null,
+  })
+  const { pb, user } = usePocketBase()
+
+  const [submitError, setSubmitError] = useState<null | string>(null)
+
+  type Errors = Partial<Record<keyof Inputs, string>>
+  const [errors, setErrors] = useState<Errors>(validate(inputs))
```

Comment:
À quoi ça sert d'avoir `errors` dans un state, vu que tu fais toujours `setInputs` en même que `setErrors`.  
Tu peux pas juste :
```suggestion
  const errors = validate(inputs)
```